frontend/components/TooltipWrapper/TooltipWrapper.tsx

```
@@ -1,9 +1,83 @@
+      if (typeof range.getClientRects !== "function") return;
+      const rects = range.getClientRects();
+      // Range.getClientRects returns one rect per text run per line, so a line
+      // containing text plus a nested <strong>/<em>/<b> produces multiple
+      // narrower rects. Taking the widest single rect would under-measure the
+      // line width. Group rects by their top edge (visual line) and compute
+      // each line's true width from the leftmost/rightmost extents, then pick
+      // the widest line.
+      const lineBounds = new Map<number, { left: number; right: number }>();
+      for (let i = 0; i < rects.length; i += 1) {
+        const rect = rects[i];
+        if (rect.width === 0) continue;
```

ESLint

[error] Unexpected use of continue statement.

(no-continue)